### PR TITLE
AN-2540: Generate code to check admin scope

### PR DESCRIPTION
**Pull request by bong-sc**
SafetyCulture uses ASAP tokens for internal communication. The token includes "scope" to specify if it's an admin or a user request. Currently, the logic to verify that is implemented manually/repeatedly by each team. This PR proposes a solution to define the check verification in proto files and generate the logic in CI together with other permission checks.

With the change, the below code could be used to define an admin endpoint:
```proto
service ExampleWithScopes {
    rpc AdminEndpoint(Request) returns (Response) {
        option (s12.flags.permissions.required_scopes) = "admin";
    }
```

- Check https://github.com/SafetyCulture/s12-modules-go/pull/217 for API change in `sc-go.io/pkg/credentials` to support this

Other changes:
- Update the command to install `protoc-gen-go` so it will respect the version in `go.mod`
- Update the version of `google.golang.org/protobuf` to match the generated file

### CODE REVIEW

**Review by sc-peter — COMMENTED** [review]
Thanks @bong-sc, I like the facts we could use this to easily find endpoints with a certain scope based on the proto files! Looks good from a security point.

File: s12/flags/permissions/permissions.proto [hunk 1]
```diff
@@ -14,4 +14,10 @@ extend google.protobuf.MethodOptions {
   //       option (google.api.http).get = "/users/v1/profiles:GetUser"
   //     }
   repeated string required_flags = 20100;
+  // A list of required scopes for the given method.
+  // For example:
+  //     rpc Getuser(Request) returns (Response) {
+  //       option (s12.flags.permissions.required_scopes) = "admin"
+  //     }
+  repeated string required_scopes = 20101;
```
Comment (vinovarkey) [inline s12/flags/permissions/permissions.proto:22]: I have reached out to @bong-sc about dropping the `repeated` attribute from here. The reason being, we do not want to accidentally allow a user scope for an admin endpoint.

**Review by vinovarkey — COMMENTED** [review]
Added in a comment. Apart from that it looks good.

File: s12/flags/permissions/permissions.proto (hunk 1 above)
Comment (bong-sc) [reply s12/flags/permissions/permissions.proto]: Checked with @sc-peter offline, it's better to use `optional string` considering it's either an admin or a user endpoint.

**Review by vinovarkey — DISMISSED** [review]

**Review by vinovarkey — APPROVED** [review]

**Pull request merged**